### DISCUSSION TITLE
Opening external links on local computer with spaces in file names

### DISCUSSION THREAD

**Discussion by 4-Wayne** [sumatrapdfreader/sumatrapdf#2486] [Q&A]
When in a PDF doc using Sumatra and an attempt to open another file on the local computer, with spaces in the file name (or folder's names), is made by clicking on a link to that file from within the pdf, it cannot find the file because of the spaces being turned into %20.

Is there a setting that I missed that can help me?  I am working with a pdf and files linked to it with hundreds, if not thousands, of such links, so changing the file names and links is not practical at this point.    Trying to leave Adobe and Foxit, but this snafu has got me.  
Thank you

**Comment by GitHubRulesOK** [comment]
links are usually formed as URL where spaces are not allowed (thus must be converted to hex e.g. in html terms decimal32 (space) = `%20` ) that is normal for if you call files in browser across lan too 
file:///C:/intranet%20-%20edge%20policy%20IntranetFileLinksEnabled%20enabled%20does%20not%20open%20file.pdf

In order to consider if it is a fixable issue a sample needs to be tested in current pre-release, and if still a problem supplied showing the problem location

**Comment by kjk** [comment]
To investigate I need a sample document that shows the problem. See https://www.sumatrapdfreader.org/docs/How-to-submit-bug-reports

**Comment by 4-Wayne** [comment]
Thank you for your response. I'm afraid the urgency of the work I am doing compels me to use other PDF readers that work in my scenario (Adobe and Foxit seem to). I appreciate your work and your product.

**Comment by kjk** [comment]
@4-Wayne should be fixed in pre-release https://www.sumatrapdfreader.org/prerelease

But FYI: this PDF seems to be incorrectly generated. The file links in the PDF shouldn't have spaces replaced with '%20'.

**Comment by 4-Wayne** [comment]
Thank you.

Regarding the creation, LibreOffice does that, perhaps because I am using the hyperlink command, though within that it has a "local file" option so I'd assume i am using it as they designed. I have no idea why they put that there for local files.

Thanks again.

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------

**Comment by 4-Wayne** [comment]
I am using the less than perfect way of opening .odt docs linked to from within the pdf by using the open in external reader function. Is there a direct way (a setting) to open it directly in another external app before it goes to a pdf tab, when clicking on a link (like an .odt or other)?
The sample I sent has .odt links as well so you can see what i mean.

It's no big deal if not. The extra step is not a burden, unless there is already a setting to make that work directly. But I could not find one.

And have a wonderful resurrection celebration, by the way.

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------

**Comment by kjk** [comment]
Our logic is: if can open in SumatraPDF, then do that. Otherwise open in external app as defined by Windows.

.odt file is really a .zip file, which SumatraPDF can open, so we do that.

There's no setting to over-ride that.

The fastest way to open in external app is probably via Command Palette:
* `Ctrl - K` to show palette
* type `fol` to narrow down list commands
* cursor up to select `Show File In Folder` command
* `Enter` to execute command and show file in file manager
* in file manager `Enter` to invoke Window's default open action for this file type

**Comment by 4-Wayne** [comment]
Is the command palette method you described something to can somehow enter into advanced settings to make an .odt file (Libreoffice Writer document) linked to from within the pdf to open in Window's default program automatically?

I assume it was not for that purpose, just making sure I understood the suggestion properly.

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------

**Reply by kjk** [reply]
No, that's beyond capabilities of Command Palette. It's just a relatively quick way to re-open the file in a different app (you can also use menu File / Show in Folder...).

**Comment by 4-Wayne** [comment]
Thought so. Thank you.

Have a great day, evening or whatever it is in your part of the world.

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------
